tests/cases/port_test.py

```diff
@@ -56,8 +56,13 @@ class T(Task):
 
         """A test task."""
 
-        input_ports = inputs
-        output_ports = outputs
+        input_ports = {}
+        for i in inputs:
+            input_ports[i.name] = i
```

Comment:
should we make input ports a dictionary? so that we can access named port like this: 

input_ports.foo 

vs

input_ports["foo"]